src/containerapp/azext_containerapp/containerapp_env_telemetry_decorator.py

```diff
@@ -0,0 +1,163 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+from azure.cli.command_modules.containerapp._utils import safe_set, safe_get
+from knack.log import get_logger
+from knack.util import CLIError
+from ._client_factory import handle_raw_exception
+from azure.cli.command_modules.containerapp.base_resource import BaseResource
+from azure.cli.core.commands import AzCliCommand
+from typing import Any, Dict
+
+DATA_DOG_DEST = 'dataDog'
+APP_INSIGHTS_DEST = 'appInsights'
+logger = get_logger(__name__)
+
+
+class ContainerappEnvTelemetryDataDogPreviewSetDecorator(BaseResource):
+
+    def __init__(self, cmd: AzCliCommand, client: Any, raw_parameters: Dict, models: str):
+        super().__init__(cmd, client, raw_parameters, models)
+        self.managed_env_def = {}
+        self.existing_managed_env_def = {}
+
+    
+    def get_argument_open_telemetry_data_dog_site(self):
+        return self.get_param("site")
+    
+    def get_argument_open_telemetry_data_dog_key(self):
+        return self.get_param("key")
+    
+    def get_argument_enable_open_telemetry_traces(self):
+        return self.get_param("enable_open_telemetry_traces")
+    
+    def get_argument_enable_open_telemetry_metrics(self):
+        return self.get_param("enable_open_telemetry_metrics")
+
+    def construct_payload(self):
+        # General setup
+        safe_set(self.managed_env_def, "location", value=r["location"])  # required for API
+
+        # Get containerapp env properties of CA we are updating
+        try:
+            self.existing_managed_env_def = self.client.show(cmd=self.cmd, resource_group_name=self.get_argument_resource_group_name(), name=self.get_argument_name())
+        except Exception as e:
+            handle_raw_exception(e)
+
+        self.set_up_open_telemetry_data_dog_site()
+        self.set_up_open_telemetry_data_dog_key()
+        self.set_up_open_telemetry_traces_destinations()
+        self.set_up_open_telemetry_metrics_destinations()
+
```

Comment:
are these method common used for datadog/otlp?

Reply:
just for datadog